documentation/CAMARA-Security-Interoperability.md

```diff
@@ -279,6 +290,173 @@ Camara recommends that implementations run the OIDF interoperability suite and a
 * [RFC 7519 - JSON Web Token (JWT)](https://www.rfc-editor.org/info/rfc7519)
 * [RFC 7521 - Assertion Framework for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/info/rfc7521)
 * [RFC 7523 - JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/info/rfc7523)
+* [RFC 7636 - Proof Key for Code Exchange by OAuth Public Clients](https://www.rfc-editor.org/info/rfc7636)
 * [RFC 8259 - The JavaScript Object Notation (JSON) Data Interchange Format](https://www.rfc-editor.org/info/rfc8259)
 * [RFC 8414 - OAuth 2.0 Authorization Server Metadata](https://www.rfc-editor.org/info/rfc8414)
 
+## Appendix A: Error Responses
+
+This section describes the error responses that the Authorization Server MUST return for all flows defined in this profile. Based on the supported functionality, the set of errors and the scenarios that can generate them are defined from the specifications already referenced in this document.
+
+### Authentication Error Response
+
+#### Authorization Code Flow
+
+If the request fails due to a missing, invalid, or mismatching redirection URI, or if the client identifier is missing or invalid, the authorization server MUST NOT automatically redirect the user-agent and SHOULD inform the resource owner of the error. For instance, the authorization server MAY display a message to the user describing the problem.
```

Comment:
I copied the text from the standard. I think that is safer, because otherwise we might inadvertently change the standard. Specifications must be precise and reformulating specifications bears the danger of introducing misunderstandings and interpretations. The previous sentence contained the adjective "automatically" which some implementer could understand as "if I configure the authorization server" then that is not "automatic" because I configured it manually. Maybe that is contrived, but so is interpreting the specified tel:" prefix for login_hint as "Tel:" being allowed. Whatever.
 
I suggest that we copy the error response definitions verbatim to this Appendix.

Reply:
The text is exactly as it is in the [OAuth 2.0 specification](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1), with the order of the sentence changed. The only addition, as you suggested, was the new sentence regarding how the user should be informed.

Reply:
I looked at OIDC https://openid.net/specs/openid-connect-core-1_0.html#AuthError and copied the text from there.

Regarding "inform", if the user agent is a browser and the AZ returns HTTP 200 OK "information" then the resource owner would be informed. Or the AZ redirects to location: https://errors.mno.com/invalid_request_uri.
If the AZ returns HTTP 400 Bad Request { "error": "invalid_request_uri" } is the user then informed?
If the API Consumer has a mobile app, then how would it react to these cases?

I think HTTP 400 Bad Request is the cleanest error response if the redirect_uri is invalid.
Also that is my understanding of the standards OIDC and OAuth2.

I would remove the "inform" sentence.

Further reading on redirect_uri validation etc: 
https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#:~:text=stage%2C%20maybe%20because-,it,-doesn%27t%20seem%20to

Reply:
I would always quote the higher standard, meaning OIDC is based on OAuth2. So, I would quote OIDC first.

Reply:
Your strategy of referencing the higher standard may be good in many cases, but in this case the OIDC text is less precise and does not cover the case of the missing redirect_uri.

In the authserver cannot redirect scenario, the standard is clear that you must inform the resource owner, i.e. the user. So you need to communicate with them in a language they understand by displaying a message on a natural language page. Although it may seem strange to us, a human does not understand HTTP status codes, JSON format or OAuth error codes. ;)

Reply:
OIDC is more precise "If the Redirection URI is invalid, the Authorization Server MUST NOT redirect the User Agent to the invalid Redirection URI. " While OAuth2 says MUST not automatically, which would allow an error page with a button which the user might click and be tricked into a malicious URL.

My "suggestion" above follows OIDC 3.1.26 and is more strict.

Reply:
If the main concern is with the use of the word "automatically," it can be removed from the text. However, I think the rest of the paragraph in my proposal is more comprehensive as it includes the missing and mismatching `redirect_uri` parameter.